app.rb

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env ruby
+
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+class App
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  def select_opt
+    option = gets.chomp.to_i
+    case option
+    when 1 then list_books
+    when 2 then list_people
+    when 3 then create_person
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_rentals
+    else
+      puts 'Invalid number, please try again!'
+    end
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @person.each do |pers|
+      puts "[#{pers.class}] Name: #{pers.name}, ID: #{pers.id}, Age: #{pers.age}"
+    end
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+
+    print 'author:'
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+
+    puts 'Book created successfully'
+  end
+
+  def check_number(msg)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if input.is_a?(Integer) && input.positive?
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_options(msg, options)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if options.include?(input)
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def create_person
+    num = check_options('Do you want to create a student (1) or a teacher (2)? [input the number]: ', [1, 2])
+    age = check_number('Age:')
+    name = gets.chomp
+    case num
+    when 1
+      print 'Has parent permission? [y/n]:'
+      parent_permission = gets.chomp
+      case parent_permission
+      when 'y' then parent_permission = true
+      when 'n' then parent_permission = false
+      end
+      @person.push(Student.new(age, name, parent_permission: parent_permission))
+    when 2
+      print 'Specialization:'
+      specialization = gets.chomp
+      @person.push(Teacher.new(age, specialization, name))
+    else
+      puts 'Invalid number, please enter number again!'
+    end
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] Good job so far, I think it would be better if users can create a student and teacher. When I try to create a person(student or teacher) a get an error. Please kindly fix that 🙃😉 

<img width="1121" alt="Screen Shot 2022-11-10 at 17 46 22" src="https://user-images.githubusercontent.com/74501523/201144867-fcb80d5e-5640-48e5-8824-b519790f7d81.png">

- [ ] Also I will suggest you print the text `name:` when the user creates a person, so then after entering the age the users need to know that name is the next. They can't imagine that after `Age:` is name 🙃🙃